Source/WebCore/page/scrolling/ScrollingTree.cpp

```diff
@@ -465,7 +465,9 @@ bool ScrollingTree::updateTreeFromStateNodeRecursive(const ScrollingStateNode* s
     }
 
     auto nodeID = stateNode->scrollingNodeID();
-    auto parentNodeID = state.frameHostingNode ? state.frameHostingNode->scrollingNodeID() : stateNode->parentNodeID();
+    auto parentNodeID = stateNode->parentNodeID();
+    if (!parentNodeID && state.frameHostingNode)
+        parentNodeID = state.frameHostingNode->scrollingNodeID();
```

Comment:
This logic should only run if `stateNode` is a `ScrollingTreeFrameScrollingNode`, right? I'd like to see a `dynamicDowncast<>` here with that check.

Reply:
What would the check do? Not all `ScrollingTreeFrameScrollingNode` inside a remote subtree should be parented to the hosting node, I think we could have them for nested same-origin iframes.

The check currently is looking for a root node within a remote subtree. I can add an `is<ScrollingTreeFrameScrollingNode` to that, but I don't think it changes much.

Reply:
Added that!